src/main/java/edu/iastate/ece/sd/sdmay2126/runner/selenium/FBASeleniumApplicationExecutor.java

```diff
@@ -16,9 +16,15 @@
 
     @Override
     public void executeApplication(Job job) {
-        new WebDriverWait(driver, Duration.ofSeconds(10))
-                .until(d -> d.findElements(By.cssSelector("button[data-button='runApp']")))
-                .get(2) // FBA is the 3rd application on the page, thus the 3rd run button (0-indexed)
-                .click();
+        try {
+            // Wait until the run button is enabled after a reset.
+            Thread.sleep(2000);
+            new WebDriverWait(driver, Duration.ofSeconds(10))
+                    .until(d -> d.findElements(By.cssSelector("button[data-button='runApp']")))
+                    .get(2) // FBA is the 3rd application on the page, thus the 3rd run button (0-indexed)
+                    .click();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
I think we want to throw this Exception up for the runner to handle, since it means the executor isn't able to complete its step successfully.

Reply:
I think Intellij just auto threw that in when I added the try catch. Good catch.